source/Mlos.Python/mlos/OptimizerEvaluationTools/OptimizerEvaluator.py

```diff
@@ -0,0 +1,220 @@
+            optimum_definition=OptimumDefinition.UPPER_CONFIDENCE_BOUND_FOR_OBSERVED_CONFIG,
+            alpha=0.01
+        )
+
+        optima_over_time["lcb_99"] = OptimumOverTime(
+            optimization_problem=self.optimizer.optimization_problem,
+            optimum_definition=OptimumDefinition.LOWER_CONFIDENCE_BOUND_FOR_OBSERVED_CONFIG,
+            alpha=0.01
+        )
+
+        #####################################################################################################
+        i = 0
+        try:
+            with traced(scope_name="optimization_loop"):
+                for i in range(self.optimizer_evaluator_config.num_iterations):
+                    parameters = self.optimizer.suggest()
+                    objectives = self.objective_function.evaluate_point(parameters)
+                    self.optimizer.register(parameters.to_dataframe(), objectives.to_dataframe())
+
+                    if i % self.optimizer_evaluator_config.evaluation_frequency == 0:
+
+                        if self.optimizer_evaluator_config.include_pickled_optimizer_in_report:
+                            evaluation_report.add_pickled_optimizer(iteration=i, pickled_optimizer=pickle.dumps(self.optimizer))
+
+                        with traced(scope_name="evaluating_optimizer"):
+                            print(f"[{i+1}/{self.optimizer_evaluator_config.num_iterations}]")
+                            if self.optimizer.trained:
+                                gof_metrics = self.optimizer.compute_surrogate_model_goodness_of_fit()
+                                regression_model_fit_state.set_gof_metrics(data_set_type=DataSetType.TRAIN, gof_metrics=gof_metrics)
+
+                            for optimum_name, optimum_over_time in optima_over_time.items():
+                                try:
+                                    optimum_config, optimum_value = self.optimizer.optimum(
+                                        optimum_definition=optimum_over_time.optimum_definition,
+                                        alpha=optimum_over_time.alpha
+                                    )
+                                    optima_over_time[optimum_name].add_optimum_at_iteration(iteration=i, optimum_config=optimum_config, optimum_value=optimum_value)
+                                except ValueError as e:
+                                    print(e)
+                evaluation_report.success = True
+
+        except Exception as e:
+            evaluation_report.success = False
+            evaluation_report.exception = e
+            evaluation_report.exception_traceback = traceback.format_exc()
+
+        if self.optimizer.trained:
+            gof_metrics = self.optimizer.compute_surrogate_model_goodness_of_fit()
+            regression_model_fit_state.set_gof_metrics(data_set_type=DataSetType.TRAIN, gof_metrics=gof_metrics)
+
+        for optimum_name, optimum_over_time in optima_over_time.items():

```

Comment:
Add a comment explaining that we are doing one last evaluation after all optimization is done.